### ISSUE TITLE
Panic on close last buffer

### ISSUE THREAD

**Issue by PushUpek** [helix-editor/helix#2367]
### Summary

Bug on closing buffers. Helix crashes with panic error:
```thread 'main' panicked at 'called `Option::unwrap()` on a `None` value', helix-view/src/editor.rs:630:51```

### Reproduction Steps

 

I tried this:

1. `hx`
2. open file with (space + f)
3. open another file (space + f)
4. switch buffer to first file (space + b) 
5. use command `:bc` to close file
6. use command `:bc` to close last file

I expected this to happen:
Helix should remain open with an empty buffer.

Instead, this happened:
Helix crashes with panic error

### Helix log

```
thread 'main' panicked at 'called `Option::unwrap()` on a `None` value', helix-view/src/editor.rs:630:51
stack backtrace:
   0: _rust_begin_unwind
   1: core::panicking::panic_fmt
   2: core::panicking::panic
   3: helix_view::editor::Editor::replace_document_in_view
   4: helix_view::editor::Editor::close_document
   5: helix_term::commands::typed::buffer_close
   6: helix_term::commands::typed::command_mode::{{closure}}
   7: <helix_term::ui::prompt::Prompt as helix_term::compositor::Component>::handle_event
   8: helix_term::compositor::Compositor::handle_event
   9: helix_term::application::Application::handle_terminal_events
  10: helix_term::application::Application::event_loop::{{closure}}
  11: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
  12: tokio::park::thread::CachedParkThread::block_on
  13: tokio::runtime::thread_pool::ThreadPool::block_on
  14: tokio::runtime::Runtime::block_on
  15: hx::main
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
```

### Platform

macOS

### Terminal Emulator

iterm2 build: 3.4.15

### Helix Version

helix 22.05-dev (2c60798b)

**Comment by archseer** [comment]
Probably a result of https://github.com/helix-editor/helix/commit/52f5a4228aef2d31cf20aeee2bed317ba1767c18

\cc @matoous

**Comment by stjepangolemac** [comment]
Just updated to the latest release and this started happening.

```
thread 'main' panicked at 'called `Option::unwrap()` on a `None` value', helix-view/src/editor.rs:633:51
```

**Comment by the-mikedavis** [comment]
#2646 shows another way to provoke this: `:bufffer-close-all`

**Comment by gavynriebau** [comment]
I just confirmed using `git bisect` and the commands described by @CptPotato for replicating the bug that the issue did indeed first appear in [52f5a42](https://github.com/helix-editor/helix/commit/52f5a4228aef2d31cf20aeee2bed317ba1767c18).

**Comment by gavynriebau** [comment]
Added PR to fix this #2658

**Comment by archseer** [comment]
https://github.com/helix-editor/helix/pull/2658
